src/lib.rs

```diff
@@ -126,12 +129,12 @@ where
         tail: &mut Peekable<&mut Inputs>,
         rbp: Precedence,
     ) -> result::Result<Self::Output, PrattError<Self::Input, Self::Error>> {
-        if let Some(head) = tail.next() {
-            let info = self.query(&head).map_err(PrattError::UserError)?;
+        if let Some(head) = tail.next().clone() {
```

Comment:
I cloned here to satisfy the borrow checker. I don't mind having the overhead for my usecase.